drivers: serial: Fix build on NXP LPUART when ASYNCH is enabled

mmahadevan108 opened the pull request:
ASYNCH implementation in the NXP LPUART driver requires DMA. Add a conditional check to enable this feature only when DMA is neabled.

Fixes https://github.com/zephyrproject-rtos/zephyr/issues/96158

decsny commented on drivers/serial/Kconfig.mcux_lpuart:
```diff
@@ -10,7 +10,7 @@ config UART_MCUX_LPUART
 	select SERIAL_HAS_DRIVER
 	select SERIAL_SUPPORT_INTERRUPT
-	select SERIAL_SUPPORT_ASYNC
+	select SERIAL_SUPPORT_ASYNC if DT_HAS_NXP_MCUX_EDMA_ENABLED
```
there is a lot more DMAs than EDMA, this is not what we want to be doing

mmahadevan108 replied:
I check the curret dts files. All SOC's that use the `nxp,lpuart` uses `nxp,mcux-edma`. However this change does not take into account future variations.

Commit:
drivers: lpuart: Dont say we support async without DMA

The dma has to be enabled on the platform in order for ASYNC API to
work. This can be indicated by whether or not any LPUART node has the
`dmas` property set.

Signed-off-by: Declan Snyder <declan.snyder@nxp.com>
